src/lnp/application/payment/types.rs

```diff
@@ -256,3 +261,134 @@ impl DumbDefault for TempChannelId {
         Self(Default::default())
     }
 }
+
+#[derive(Wrapper, Clone, Debug, From, PartialEq, Eq)]
+pub struct NodeColor([u8; 3]);
+
+impl LightningEncode for NodeColor {
+    fn lightning_encode<E: io::Write>(
+        &self,
+        mut e: E,
+    ) -> Result<usize, io::Error> {
+        let len = e.write(self.as_inner())?;
+        Ok(len)
+    }
+}
+
+impl LightningDecode for NodeColor {
+    fn lightning_decode<D: io::Read>(mut d: D) -> Result<Self, LightningError> {
+        let mut buf = [0u8; 3];
+        d.read_exact(&mut buf)?;
+        Ok(Self::from_inner(buf))
+    }
+}
+
+#[derive(Wrapper, Clone, Debug, From, PartialEq, Eq)]
+pub struct Alias([u8; 3]);
+
+impl LightningEncode for Alias {
+    fn lightning_encode<E: io::Write>(
+        &self,
+        mut e: E,
+    ) -> Result<usize, io::Error> {
+        let len = e.write(self.as_inner())?;
+        Ok(len)
+    }
+}
+
+impl LightningDecode for Alias {
+    fn lightning_decode<D: io::Read>(mut d: D) -> Result<Self, LightningError> {
+        let mut buf = [0u8; 3];
+        d.read_exact(&mut buf)?;
+        Ok(Self::from_inner(buf))
+    }
+}
+
+/// Lightning network short channel Id as per BIP7
+#[derive(
+    Clone,
+    Copy,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    Debug,
+    Default,
+    From,
+    StrictEncode,
+    StrictDecode,
+)]
+#[lnpbp_crate(crate)]
+pub struct ShortChannelId {
+    pub block_height: u32,
+    pub tx_index: u32,
+    pub output_index: u16,
+}
```

Comment:
We do not check that `block_height` and `tx_index` are `<2^24`. What I propose:

1. Set all fields to be private
2. do `derive(Getters)` for accessing them
3. Provide a constructor that returns optional with `None` value if the provided values does not fit `0..2^24` range

Reply:
Added